core/src/main/java/hudson/model/Queue.java

```diff
@@ -866,7 +841,7 @@ private WaitingItem peek() {
      * @since 1.483
      */
     public List<Item> getApproximateItemsQuickly() {
-        return itemsView.get();
+        return getItemsAsList();
```

Comment:
The `getItems` builds a list to convert it to an array. `Arrays.asList` would them wrap it in a `List`. What we do here is move the list building to an auxiliary method and reuse the result in `getApproximateItemsQuickly`.

The net balance is the same and we are returning exactly the same object (`ImmutableList`) which is more appropriate to have if, in the future, we feel again the need for the cache if permission checking turns to be a potential bottleneck (authorization strategies can get big).